### PR TITLE
added library/cereal

### PR DESCRIPTION
**Pull request by erwinlem**
This is a dependency for Domoticz, will do a pull request for that once merged.

### CODE REVIEW

File: components/library/cereal/Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+COMPONENT_ARCHIVE=      $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_URL=	https://github.com/USCiLab/cereal/archive/refs/tags/v$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:16a7ad9b31ba5880dac55d62b5d6f243c3ebc8d46a3514149e56b5e7ea81f85f
+COMPONENT_LICENSE=	BSD3CLAUSE
+COMPONENT_LICENSE_FILE=	LICENSE
+
+TEST_TARGET=$(NO_TESTS) # if no testsuite enabled
+include $(WS_MAKE_RULES)/common.mk
+
+CMAKE_OPTIONS+= -DBUILD_TESTS=OFF
```
Comment (AndWac) [inline components/library/cereal/Makefile:34]: Did you run "gmake REQUIRED_PACKAGES"? If so, there should be at least a generated comment here.

File: components/library/cereal/Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+BUILD_STYLE=cmake
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		cereal
+COMPONENT_VERSION=	1.3.2
+COMPONENT_SUMMARY=	A C++11 library for serialization
+COMPONENT_PROJECT_URL=	https://uscilab.github.io/cereal/
+COMPONENT_FMRI=		library/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION=
+COMPONENT_SRC=          $(COMPONENT_NAME)-$(COMPONENT_VERSION)
```
Comment (AndWac) [inline components/library/cereal/Makefile:24]: Please add a COMPONENT_CLASSIFICATION

File: components/library/cereal/Makefile (hunk 1 above)
Comment (erwinlem) [reply components/library/cereal/Makefile:34]: I did, it was empty. Added the separator so it's clear I did.

**Comment by AndWac** [comment]
This runs into a build error quite early:
CMake Error at /usr/lib/cmake/Boost-1.78.0/BoostConfig.cmake:141 (find_package):
  Found package configuration file:

    /usr/lib/cmake/boost_serialization-1.78.0/boost_serialization-config.cmake

  but it set boost_serialization_FOUND to FALSE so package
  "boost_serialization" is considered to be NOT FOUND.  Reason given by
  package:

  No suitable build variant has been found.

  The following variants have been tried and rejected:

  * libboost_serialization.so.1.78.0 (32 bit, need 64)

  * libboost_serialization.a (32 bit, need 64)

Call Stack (most recent call first):
  /usr/lib/cmake/Boost-1.78.0/BoostConfig.cmake:262 (boost_find_component)
  /usr/share/cmake-3.23/Modules/FindBoost.cmake:594 (find_package)
  CMakeLists.txt:111 (find_package)

**Comment by erwinlem** [comment]
I removed 32 bit support on my version of Boost, because i ran into thesame problem with porting Ceph. For some reason cmake finds the 32bit libraries instead of 64bit.  BoostConfig.cmake has a 32 and 64 bit version, cmake picks the wrong one for some reason. What is the proper way to solve this?

For now i disabled the performance tests which "solves" the problem.

**Comment by AndWac** [comment]
We have a similar problem with other packages, too. We need to find a way to teach cmake to use the 64-bit library.

**Comment by erwinlem** [comment]
CMAKE_LIBRARY_ARCHITECTURE=amd64 is the proper way to fix this. That changes the search path for cmake. I think this should be set by default for all cmake amd64 builds.

I also noticed in cmake.mk that LDFLAGS & PKG_CONFIG_PATH are added to options and not to the environment. Was that a mistake? It does generate warnings and makes no sense.

**Comment by AndWac** [comment]
You are most probably right. Can you propose fixes or enhancements and create a PR with them, please?

File: components/library/cereal/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+COMPONENT_ARCHIVE_URL=	https://github.com/USCiLab/cereal/archive/refs/tags/v$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:16a7ad9b31ba5880dac55d62b5d6f243c3ebc8d46a3514149e56b5e7ea81f85f
+COMPONENT_LICENSE=	BSD3CLAUSE
+COMPONENT_LICENSE_FILE=	LICENSE
+
+TEST_TARGET=$(NO_TESTS) # if no testsuite enabled
+include $(WS_MAKE_RULES)/common.mk
+
+CMAKE_OPTIONS+= -DCMAKE_LIBRARY_ARCHITECTURE=amd64
+CMAKE_OPTIONS+= -DBUILD_TESTS=OFF
```
Comment (AndWac) [inline components/library/cereal/Makefile:34]: Do you want to remove now that after your cmake.mk PR has been merged?

**Pull request merged**